src/Symfony/Component/Security/Password/SaltingPasswordHasherInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Password;
+
+use Symfony\Component\Security\Password\Exception\InvalidPasswordException;
+
+/**
+ * Provides password hashing and verification capabilities with salting.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+interface SaltingPasswordHasherInterface extends PasswordHasherInterface
```

Comment:
why not `SelfSaltingPasswordHasherInterface`?

Also, would there be a way to remove this interface and merge it with `PasswordHasherInterface`? Aka remove the concept of salt from our interfaces - since it's an obsolete one?

Reply:
This one does the inverse of the current `SelfSaltingEncoderInterface`: it is should only be implemented by implementations that need user-provided salts, so that modern hashers (native, sodium etc) don't take a no-op `salt` argument.
Removing the concept entirely would be hard since we still have old hashers that require a salt... so the idea is to have this interface until we support those old hashers

Reply:
but `SaltingPasswordHasherInterface` is a bad name for such hasher. Modern hashers are also salting the password, but they do that without requiring an external salt.

Reply:
`LegacyPasswordHasherInterface` to make this clear?

Reply:
👍 thanks, renamed